src/components/calcite-dropdown-item/calcite-dropdown-item.tsx

```diff
@@ -53,6 +53,12 @@ export class CalciteDropdownItem {
   /** pass an optional title for rendered href */
   @Prop() linkTitle?: string;
 
+  /** optionally pass an icon to display at the start of an item - accepts calcite ui icon names  */
+  @Prop({ reflect: true }) iconStart?: string;
+
+  /** optionally pass an icon to display at the end of an item - accepts calcite ui icon names  */
+  @Prop({ reflect: true }) iconEnd?: string;
```

Comment:
Are you sure you don't want to go the slot route? What if someone then asks for the ability to have a menu on one icon?

Reply:
I think that it's a good balance and will be consistent with how buttons and eventually inputs accept icons - this should take care of consistently styling the majority of use cases for using icons in dropdown items, if there's something more specific to an app like... an overflow menu in a dropdown they still can accomplish that, the styling of that is just on the implementer:

```
<calcite-dropdown-item>
    Item text
    <div class="my-custom-styled-icon-and-overflow-menu">
    ...
    </div>
</calcite-dropdown-item>
```